### PR TITLE
Add references to objects when creating iterators

### PR DESCRIPTION
**Pull request by rtheunissen**
See #82

### CODE REVIEW

File: src/php/iterators/php_htable_iterator.h [hunk 1]
```diff
@@ -10,6 +10,7 @@ typedef struct ds_htable_iterator {
     uint32_t                 position;
     ds_htable_bucket_t      *bucket;
     ds_htable_t             *table;
+    zval                    *obj;
```
Comment (nikic) [inline src/php/iterators/php_htable_iterator.h]: This should be a `zval` into which you ZVAL_COPY, or a `zend_object*` (in with case you can use GC_ADDREF and OBJ_RELEASE as the refcount operations). You can't keep just a pointer here, as the lifetime of the iterator may exceed the lifetime of that zval.

Comment (rtheunissen) [inline src/php/iterators/php_htable_iterator.h]: Can do either easily enough. Any preference for zval vs zend_object* here or is the result the same? The current patch passed the tests without leaks but I see what you're saying here. Will need to do the same with the sequence iterators too.

Comment (nikic) [inline src/php/iterators/php_htable_iterator.h]: The result will be the same, zend_object* is just a bit more compact. (Not that it matters here.)

Comment (rtheunissen) [inline src/php/iterators/php_htable_iterator.h]: I can't seem to find any references to `GC_ADDREF` in php-src (can't find a working lxr either). Is this basically just `++GC_REFCOUNT(zend_object_ptr)`?

Comment (rtheunissen) [inline src/php/iterators/php_htable_iterator.h]: @nikic Something like this? See adc912998bf25a569c3d392b5cbcffeb9281ba0b

**Comment by rtheunissen** [comment]
I'm happy with this @nikic, thanks for your help. ✨ 
Will leave it open for a while but will merge if there's no further suggestion or warning.

**Comment by nikic** [comment]
lgtm

File: src/php/iterators/php_htable_iterator.h (hunk 1 above)
Comment (staabm) [inline src/php/iterators/php_htable_iterator.h]: fyi https://lxr.room11.org/

Comment (rtheunissen) [inline src/php/iterators/php_htable_iterator.h]: Thanks @staabm!

**Pull request merged**